concordium-consensus/src/Concordium/Scheduler/TreeStateEnvironment.hs

```diff
@@ -332,9 +421,449 @@ doBlockReward transFees FreeTransactionCounts{..} bid foundationAddr bs0 = do
       stoFoundationAccount = foundationAddr
     }
 
+-- |Accrue the rewards for a block to the relevant pool, the L-pool, and the foundation.
+doBlockRewardP4 :: forall m. (BlockStateOperations m, MonadProtocolVersion m, ChainParametersVersionFor (MPV m) ~ 'ChainParametersV1, AccountVersionFor (MPV m) ~ 'AccountV1)
+  => Amount
+  -- ^Transaction fees paid
+  -> FreeTransactionCounts
+  -- ^Counts of unpaid transactions
+  -> BakerId
+  -- ^Block baker
+  -> UpdatableBlockState m
+  -- ^Block state
+  -> m (UpdatableBlockState m)
+doBlockRewardP4 transFees FreeTransactionCounts{..} bid bs0 = do
```

Comment:
Wouldn't it be more readable to write this as follows

```haskell
bsoSetRewardAccounts args
>>= bsoUpdateAccruedTransactionFeesFoundationAccount args
>>= bsoUpdateAccruedTransactionFeesLPool args
>>= bsoUpdateAccruedTransactionFeesBaker args
>>= bsoAddSpecialTransactionOutcome args
```
?

That would require changing the argument order in the block state operations, but not having to look up variable names helps readability.